### PR TITLE
Head of meta

### PR DESCRIPTION
**Pull request by allardy**

### CODE REVIEW

File: packages/studio-ui/src/web/translations/fr.json
Comment (franklevasseur) [inline packages/studio-ui/src/web/translations/fr.json:310]: contient

File: packages/studio-ui/src/web/views/FlowBuilder/sidePanelFlows/FlowNameModal.tsx
Comment (franklevasseur) [inline packages/studio-ui/src/web/views/FlowBuilder/sidePanelFlows/FlowNameModal.tsx:64]: This feels a little naked. Maybe there are better places for this condition to be then hardcoded in the UI. 

Then again, I don't really mind

**Review by franklevasseur — COMMENTED** [review]
**fix(studio): prevent using skills in folde rname**

Works well ✅

File: packages/studio-ui/src/web/views/FlowBuilder/sidePanelFlows/FlowNameModal.tsx
Comment (allardy) [reply packages/studio-ui/src/web/views/FlowBuilder/sidePanelFlows/FlowNameModal.tsx:64]: I thought of implementing something better, but since it's the only case we have to handle, i prefer to not spend more time. Thought of an array of illegal names, regex, etc, but didnt wanna overengineer for this specific case

File: packages/studio-be/src/core/bots/bot-service.ts
Comment (franklevasseur) [inline packages/studio-be/src/core/bots/bot-service.ts:289]: This is so hackish...

There's a lot of ways of doing this, but this one just feels off. I personally would do something like:

**solution 1**
```ts
class BotService {
    public listenForBotUnmount(listener: (botId: string) => Promise<void>) {}
}

class FlowService {
    constructor(private botService: BotService) {
        botService.listenForBotUnmount(this.handleUnmount.bind(this))
    }
    
    private handleUnmount(botId: string) { delete this.scopes[botId] }
}
```

**solution 2**
```ts
class BotService {
    constructor(private flowService: FlowService) {}
    
    public unmountBot(botId: string) {
        this.flowService.removeFlowsForBot(botId)
    }
}

class FlowService { 
    public removeFlowsForBot(botId: string) { delete this.scopes[botId] }
}
```

**Review by franklevasseur — COMMENTED** [review]
**fix(studio): no ghost flows when deleting a bot**

Tested and it works ✅

File: packages/studio-ui/src/web/views/FlowBuilder/sidePanelFlows/FlowNameModal.tsx
Comment (franklevasseur) [reply packages/studio-ui/src/web/views/FlowBuilder/sidePanelFlows/FlowNameModal.tsx:64]: makes sense

File: packages/studio-ui/src/web/translations/fr.json
Comment (franklevasseur) [inline packages/studio-ui/src/web/translations/fr.json:365]: We use "node" everywhere in translations

File: packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.jsx
Comment (franklevasseur) [inline packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.jsx:14]: There a lot of unnecessary indentation in this function:

```ts
if (!text) {
    return toast.failure(lang.tr('studio.flow.node.emptyName'))
}

if (text !== this.props.node.name) {
    return
}

const alreadyExists = this.props.flow.nodes.find(x => x.name === text)
if (alreadyExists) {
    return toast.failure(lang.tr('studio.flow.node.nameAlreadyExists'))
}

this.props.updateNode({ name: text })
```

File: packages/studio-ui/src/web/views/FlowBuilder/sidePanelTopics/form/SaySomethingForm.tsx
Comment (franklevasseur) [inline packages/studio-ui/src/web/views/FlowBuilder/sidePanelTopics/form/SaySomethingForm.tsx:55]: What's this class ? Why is this dupplicated ? 

Same comment goes for here, there's a lot of indentation

**Review by franklevasseur — COMMENTED** [review]
**fix(studio): fix error when renaming node**

tested and it works @sebburon ✅

File: packages/studio-ui/src/web/components/ContentForm/UploadWidget/UrlUpload.tsx
Comment (franklevasseur) [inline packages/studio-ui/src/web/components/ContentForm/UploadWidget/UrlUpload.tsx:46]: This doesnt check if braces are closed, but its good enough for this feature
\+ it will display warning as soon as user types '{{'

**Review by franklevasseur — COMMENTED** [review]
**fix(studio): warns user about double braces** ❌

I can't seem to find the warning anywhere in studio:
image

@ptrckbp tell me If I'm missing something here

File: packages/studio-ui/src/web/components/Authentication/index.jsx
Comment (franklevasseur) [inline packages/studio-ui/src/web/components/Authentication/index.jsx:36]: hackish but it works

**Review by franklevasseur — COMMENTED** [review]
**fix(studio): Warns user he will be logged out**

Tested and it works ✅

Let's just make sure with @J-FMartin that he agrees with this being in the product

**Comment by franklevasseur** [comment]
Allright, here's what I got
| commit                                           | author | Points          | PR                                                  | Issue                                                     | Bug Reproduced | Code Reviewed | Bug Fixed |
| ------------------------------------------------ | ------ | --------------- | --------------------------------------------------- | --------------------------------------------------------- | -------------- | ------------- | --------- |
| fix(studio): prevent using skills in folde rname | yann   | 1               | [#157](https://github.com/botpress/studio/pull/157) | [#43](https://github.com/botpress/studio/pull/43)         | yes            | yes           | yes       |
| fix(studio): no ghost flows when deleting a bot  | yann   | 2               | [#160](https://github.com/botpress/studio/pull/160) | [#5393](https://github.com/botpress/botpress/pull/5393)   | yes            | yes           | yes       |
| fix(studio): fix error when renaming node        | seb    | 1               | [#159](https://github.com/botpress/studio/pull/159) | [#52](https://github.com/botpress/studio/pull/52)         | yes            | yes           | yes       |
| fix(studio): warns user about double braces      | pat    | no estimate (1) | [#168](https://github.com/botpress/studio/pull/168) | [#5370](https://github.com/botpress/botpress/issues/5370) | yes            | yes           | yes       |
| fix(studio): Warns user he will be logged out    | pat    | no estimate (1) | [#185](https://github.com/botpress/studio/pull/185) | [#5272](https://github.com/botpress/botpress/pull/5272)   | yes            | yes           | yes       |

@allardy, you're the code owner, feel free to check the comments I left or not. The code works.

Before merging:
1. Check with JF that he agrees with **fix(studio): Warns user he will be logged out** #185
2. Make sure sure not to squash

**Review by EFF — APPROVED** [review]
LGTM

**Pull request merged**
